tests/clone-flags.c

```diff
@@ -23,11 +23,39 @@
 static const int child_exit_status = 42;
 static pid_t pid;
 
+static void
+retrieve_userns(pid_t pid, char *userns_buf, size_t userns_buflen)
+{
+	char *fname = xasprintf("/proc/%d/ns/user", pid);
+	ssize_t rc = readlink(fname, userns_buf, userns_buflen - 1);
+	if (rc < 0) {
+                perror(fname);
+                userns_buf[0] = '\0';
+        } else if ((size_t) rc >= userns_buflen)
+                error_msg_and_fail("readlink: %s: result is too long", fname);
+        else
+                userns_buf[rc] = '\0';
+	userns_buf[rc] = '\0';
```

Comment:
`userns_buf[rc] = '\0'` is already done where necessary, and this one is harmful when rc < 0.